src/kili/services/json_response/annotation.py

```diff
@@ -0,0 +1,262 @@
+        """
+        return self.json_data["score"]
+
+
+class BoundingPolyAnnotation(_Base2DAnnotation):
+    """Class for parsing the "annotations" key of a job response for 2D object detection jobs."""
+
+
+class VideoAnnotation(_Base2DAnnotation):
+    """Class for parsing the "annotations" key of a job response for video object detection jobs."""
+
+    @property
+    def is_key_frame(self) -> bool:
+        """Returns a boolean indicating if the timestamp or frame is used for interpolation."""
+        try:
+            return self.json_data["isKeyFrame"]
+        except KeyError as err:
+            raise AttributeNotCompatibleWithJobError("is_key_frame") from err
+
+
+class PoseEstimationAnnotation(_BaseAnnotationWithTool):
+    """Class for parsing the "annotations" key of a job response for pose estimation jobs."""
+
+    @staticmethod
+    def _get_compatible_ml_task() -> str:
+        return "OBJECT_DETECTION"
+
+    @staticmethod
+    def _get_compatible_type_of_tools() -> Sequence[str]:
+        return ("pose",)
+
+    @property
+    def kind(self) -> str:
+        """Returns the job kind. In pose estimation jobs, this is always "POSE_ESTIMATION"."""
+        return self.json_data["kind"]
+
+    @property
+    def points(self) -> List[Dict]:
+        """Returns the list of the points composing the object."""
+        return self.json_data["points"]
+
+
+def check_attribute_compatible_with_job(func):
+    """Raises an error if the decorated method is not compatible with the job interface mlTask."""
+
+    @functools.wraps(func)
+    # pylint: disable=protected-access
+    def wrapper(self, *args, **kwargs):
+        attribute_name = func.__name__
+
+        attr_comp_with_mltask = (
+            attribute_name in self._valid_attributes_for_ml_task[self.job_interface["mlTask"]]
+        )
+        if not attr_comp_with_mltask:
+            raise AttributeNotCompatibleWithJobError(attribute_name)
+
+        if "type" in self.json_data:
+            attr_comp_with_tool = (
+                attribute_name in self._valid_attributes_for_tool[self.json_data["type"]]
+            )
+            if not attr_comp_with_tool:
+                raise AttributeNotCompatibleWithJobError(attribute_name)
+
+        return func(self, *args, **kwargs)
+
+    return wrapper
+
+
+@for_all_properties(check_attribute_compatible_with_job)
+# pylint: disable=too-many-ancestors
+class Annotation(
+    EntityAnnotation,
+    PointAnnotation,
+    BoundingPolyAnnotation,
+    VideoAnnotation,
+    PoseEstimationAnnotation,
+):
+    """Class for parsing the "annotations" key of a job response.
+
+    Contains all attributes that can be found in a job response.
+    """
+
+    def __init__(self, json_data: Dict, job_interface: Dict) -> None:
+        """Initializes an Annotation object.
+
+        This class is used to parse the "annotations" key of a job response.
+
+        Args:
+            json_data (Dict): The json data of the annotation.
```

Comment:
no need to specify the argument types (already in the signature, cf https://google.github.io/styleguide/pyguide.html#383-functions-and-methods)